File: Serenity.Script.UI/SlickGrid/Formatters/MinuteFormatter.cs
using jQueryApi;
using Serenity.ComponentModel;
using System;

namespace Serenity
{
    public class MinuteFormatter : ISlickFormatter
    {
        public string Format(SlickFormatterContext ctx)
        {
            return Format(ctx.Value);
        }

        public static string Format(object value)  // 730 = 12:10
        {
            int hour = (int)Math.Floor(value.As<double>()/60);
            int minute = value.As<Int32>() - (hour * 60);

            string hourStr, minuteStr;

            if (!Script.IsValue(value) ||
                Double.IsNaN(value.As<double>()))
                return "";

            if (hour < 10)
                hourStr = "0" + hour;
            else
                hourStr = hour.ToString();

            if (minute < 10)
                minuteStr = "0" + minute;
            else
                minuteStr = minute.ToString();
            
            return String.Format("{0}:{1}", hourStr, minuteStr);
        }
    }
}

File: packages/corelib/src/ui/formatters/minuteformatter.ts
import { FormatterContext } from "@serenity-is/sleekgrid";
import { formatterTypeInfo, nsSerenity, registerType, stringFormat } from "../../base";
import { Formatter } from "../../slick";

/** Formats an integer minute count as `HH:mm` (e.g. 90 → `"01:30"`). */
export class MinuteFormatter implements Formatter {
    static [Symbol.typeInfo] = formatterTypeInfo(nsSerenity); static { registerType(this); }

    /**
     * Formats the cell value as `HH:mm`.
     * @param ctx - Formatter context containing the minute value.
     * @returns `HH:mm` string.
     */
    format(ctx: FormatterContext) {
        return MinuteFormatter.format(ctx.value);
    }

    /**
     * Static helper to format minutes.
     * @param value - Total minutes.
     * @returns `HH:mm` string or empty if invalid.
     */
    static format(value: number): string {
        if (value == null || (value as any) === '' || !isFinite(value))
            return '';

        var hour = Math.floor(value / 60);
        var minute = value - hour * 60;
        var hourStr, minuteStr;

        if (hour < 10)
            hourStr = '0' + hour;
        else
            hourStr = hour.toString();

        if (minute < 10)
            minuteStr = '0' + minute;
        else
            minuteStr = minute.toString();

        return stringFormat('{0}:{1}', hourStr, minuteStr);
    }
}
